RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -0,0 +1,609 @@
+				      prop.product(), !mtdthits.empty(),pathLength);
+      if( result.ndof() >= 0 ) {
+        /// setup the track extras
+        reco::TrackExtra::TrajParams trajParams;
+        reco::TrackExtra::Chi2sFive chi2s; 
+        size_t hitsstart = outhits->size();
+        if( updatePattern_ ) {
+          t2t(trj,*outhits,trajParams,chi2s); // this fills the output hit collection
+        } else {
+          t2t(thetrj,*outhits,trajParams,chi2s);
+        }
+        size_t hitsend = outhits->size();
+        extras->push_back(buildTrackExtra(trj)); // always push back the fully built extra, update by setting in track
+        extras->back().setHits(hitsRefProd,hitsstart,hitsend-hitsstart);	
+        extras->back().setTrajParams(trajParams,chi2s);        
+        //create the track
+        output->push_back(result);
+	pathLengthsRaw.push_back(pathLength);
+	pathLengthMap = pathLength;
+        auto& backtrack = output->back();
+	betaMap = backtrack.beta();
+	t0Map = backtrack.t0();
+	covt0t0Map = backtrack.covt0t0();
+        reco::TrackExtraRef extraRef(extrasRefProd,extras->size()-1);
+        backtrack.setExtra( (updateExtra_ ? extraRef : track.extra()) );
+        for(unsigned ihit = hitsstart; ihit < hitsend; ++ihit) {
+          backtrack.appendHitPattern((*outhits)[ihit],ttopo);
+        }
+      }
+      pathLengthsOrigTrkRaw.push_back(pathLengthMap);
+      betaOrigTrkRaw.push_back(betaMap);
+      t0OrigTrkRaw.push_back(t0Map);
+      covt0t0OrigTrkRaw.push_back(covt0t0Map);
+    }
+    ++itrack;
+  }
+
+  auto outTrksHandle = ev.put(std::move(output));
+  ev.put(std::move(extras));
+  ev.put(std::move(outhits));
+
+  edm::ValueMap<float>::Filler fillerPathLengths(*pathLengths);
+  fillerPathLengths.insert(outTrksHandle,pathLengthsRaw.cbegin(),pathLengthsRaw.cend());
+  fillerPathLengths.fill();
+  ev.put(std::move(pathLengths),pathLengthName);
+
+  edm::ValueMap<float>::Filler fillerPathLengthsOrigTrk(*pathLengthsOrigTrk);
+  fillerPathLengthsOrigTrk.insert(tracksH,pathLengthsOrigTrkRaw.cbegin(),pathLengthsOrigTrkRaw.cend());
+  fillerPathLengthsOrigTrk.fill();
+  ev.put(std::move(pathLengthsOrigTrk),pathLengthOrigTrkName);
+
+  edm::ValueMap<float>::Filler fillerBetas(*betaOrigTrk);
+  fillerBetas.insert(tracksH,betaOrigTrkRaw.cbegin(),betaOrigTrkRaw.cend());
+  fillerBetas.fill();
+  ev.put(std::move(betaOrigTrk),betaOrigTrkName);
+
+  edm::ValueMap<float>::Filler fillert0s(*t0OrigTrk);
+  fillert0s.insert(tracksH,t0OrigTrkRaw.cbegin(),t0OrigTrkRaw.cend());
+  fillert0s.fill();
+  ev.put(std::move(t0OrigTrk),t0OrigTrkName);
+
+  edm::ValueMap<float>::Filler fillercovt0t0s(*covt0t0OrigTrk);
+  fillercovt0t0s.insert(tracksH,covt0t0OrigTrkRaw.cbegin(),covt0t0OrigTrkRaw.cend());
+  fillercovt0t0s.fill();
+  ev.put(std::move(covt0t0OrigTrk),covt0t0OrigTrkName);
+}
+
+namespace {
+  auto cmp = [](const unsigned one, const unsigned two) -> bool { return one < two; };
```

Comment:
static analysis is not happy about this.
perhaps move this inside void find_hits_in_dets ?

Reply:
Or just make it a function named `cmp`?

Reply:
Or use `std::less<>`?

Reply:
the location of this cmp is the same as before.
So, the static analyzer issue remains